data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/stream/ShardConsumer.java

```diff
@@ -260,19 +270,21 @@ public void run() {
             }
         }
 
-        if (acknowledgementSet != null) {
-            checkpointer.updateShardForAcknowledgmentWait(shardAcknowledgmentTimeout);
-            acknowledgementSet.complete();
-        }
-
         // interrupted
         if (shouldStop) {
             // Do last checkpoint and then quit
-            LOG.error("Should Stop flag is set to True, looks like shutdown has triggered");
+            LOG.warn("Processing for shard {} was interrupted by a shutdown signal, giving up shard", shardId);
```

Comment:
Should we log the giving up or completion in steam scheduler? because here it's just to stop the run, but the action after such stop is decided in scheduler.  But at least for now your change is ok.

Reply:
Yeah I did think about that but didn't want two different logs.